evm_arithmetization/src/cpu/kernel/asm/main.asm

```diff
@@ -41,10 +41,13 @@ global hash_initial_tries:
     // stack: trie_data_full_len
     %mstore_global_metadata(@GLOBAL_METADATA_TRIE_DATA_SIZE)
 
+    // If txn_idx == 0, update the beacon_root.
```

Comment:
DQ: where do we check if we are beyond the Cancun HF block?

Reply:
We don't, neither for Shanghai at the moment, i.e. clients must use the correct prover according to the HF they are currently on.